src/platforms/esp32/components/avm_builtins/gpio_driver.c

```diff
@@ -343,11 +343,32 @@ static bool gpiodriver_is_gpio_attached(struct GPIOData *gpio_data, int gpio_num
 static term gpiodriver_set_int(Context *ctx, int32_t target_pid, term cmd)
 {
     GlobalContext *glb = ctx->global;
+    int32_t target_local_pid;
 
     struct GPIOData *gpio_data = ctx->platform_data;
 
     int32_t gpio_num = term_to_int32(term_get_tuple_element(cmd, 1));
     term trigger = term_get_tuple_element(cmd, 2);
+    if (term_get_tuple_arity(cmd) == 4) {
+        term pid = term_get_tuple_element(cmd, 3);
+        if (UNLIKELY(!term_is_pid(pid) && !term_is_atom(pid))) {
+            ESP_LOGE(TAG, "Invalid listener parameter, must be a pid() or registered process!");
+            return ERROR_ATOM;
```

Comment:
Why not return an `{error, Reason}` pair like you do in the STM32 driver?  I think it would be better if we did and would be one more step towards removing the `error` return in the type spec.

Reply:
I had this waiting for a while (wanted the GPIO clenup/fixes to go in first), I just missed this in the rebase. Good catch.

Reply:
I didn't address that in this PR. Since that involves more changes outside the scope of this PR. But I agree that the esp32 driver does need to update the errors returns. I believe we want the nifs to RAISE_ERROR(Reason) and have the port functions return {error, Reason}.

Reply:
I was thinking about this again, and since this is a new function I would rather start the new convention.
So let's return the error tuple, so we avoid in advance breaking the API and boring changes later.
Adding a short note about the return type difference would be IMHO enough.